web/modules/custom/dpl_event/src/EventSubscriber/EventSeriesRedirect.php

```diff
@@ -31,7 +36,14 @@ public function __construct(
    */
   public static function getSubscribedEvents(): array {
     return [
-      KernelEvents::REQUEST => 'checkEventSeriesRedirect',
+      KernelEvents::REQUEST => [
+        ['checkEventSeriesRedirect'],
+      ],
+      // Only target users that end up on an unpublished eventinstance page,
+      // without being allowed to see it.
+      KernelEvents::EXCEPTION => [
+        ['checkEventInstanceRedirect'],
+      ],
```

Comment:
Please reconsider the structure of this addition.

As I see it you are adding new functionality to an existing class that related somewhat to the domain of the code - events and redirects. 

There is also some differences. The current class deals with event series as indicated by the name. The addition deals with event instances. There is no code reuse between the existing and new code.

This in mind I would suggest moving it to a separate class.

Reply:
I'll leave it as it is, because it is about redirection *to* and *from* series - as i write in the comments :)